single_kernel_mongo/managers/mongo.py

```diff
@@ -318,24 +318,29 @@ def update_app_relation_data_for_config(self, relation: Relation, config: MongoC
         endpoints = data_interface.fetch_my_relation_field(relation.id, "endpoints") or ""
         uris = data_interface.fetch_my_relation_field(relation.id, "uris")
         database = data_interface.fetch_my_relation_field(relation.id, "database")
-        with MongoConnection(self.state.mongo_config) as mongo:
-            user_exists = mongo.user_exists(config.username)
-        if user_exists:
-            if config.hosts != set(endpoints.split(",")):
-                data_interface.set_endpoints(
-                    relation.id,
-                    ",".join(config.hosts),
-                )
-            if config.uri != uris:
-                data_interface.set_uris(
-                    relation.id,
-                    config.uri,
-                )
-            if config.database != database:
-                data_interface.set_database(
-                    relation.id,
-                    config.database,
-                )
+        username = data_interface.fetch_my_relation_field(relation.id, "username")
+        password = data_interface.fetch_my_relation_field(relation.id, "password")
+        if not username or not password:
+            data_interface.set_credentials(
```

Comment:
will the linter complain if we put arguments in the same line?

Reply:
You mean on the `set_credentials` call ? No I don't think so but I find it easier to read this way.
No strong opinion though.